net/AsyncDNS.hpp

```diff
@@ -42,7 +43,9 @@ class AsyncDNS
     typedef std::function<void(const HostEntry& hostEntry)> DNSThreadFn;
     typedef std::function<std::string()> DNSThreadDumpStateFn;
 
-    static void lookup(const std::string& searchEntry, const DNSThreadFn& cb,
+    static void lookup(const std::string& searchEntry,
+                       const std::optional<std::string>& port,
```

Comment:
Wouldn't an empty `port` imply it's missing? That is, an empty string is already a natural sentry, instead of `std::optional`.

Edit: It's unclear from getaddrinfo(3) what happens with a non-null `service` that is empty (for `AI_CANONNAME | AI_ADDRCONFIG`). It seams it wouldn't care for those cases, but it still seems cleaner to _not_ pass in empty strings. So avoiding `optional` and checking only for empty string to pass the pointer or nullptr seems most reasonable here.

Reply:
yeah, fine by me